pandas/core/dtypes/concat.py

```diff
@@ -480,20 +480,22 @@ def _concat_datetimetz(to_concat, name=None):
 
 def _concat_index_same_dtype(indexes, klass=None):
     klass = klass if klass is not None else indexes[0].__class__
-    return klass(np.concatenate([x._values for x in indexes]))
+    return klass(np.concatenate([x._ndarray_values for x in indexes]))
```

Comment:
This one is only used for numeric indices, so `_values` or `_ndarray_values` should not matter. 
I was mainly thinking that for those cases where `_ndarray_values` actually differs from `_values`, like a categorical, the above code will not work anyway, as `klass(codes)` is not enough to reconstruct the categorical. So `_values` seems safer to me.